server/services/survey.service.js

```diff
@@ -0,0 +1,60 @@
+const Path = require('path-parser');
+const { URL } = require('url');
+
+module.exports = class SurveyService {
+
+  async updateSurveys(data) {
+    try {
+
+      const uniqueServeys =
+          data.map(this.buildSurveyDataFromBody)
+              .filter(this.filterOutNullAndUndefined)
+              .filter(this.filterByUniqueEmailAndSurveyId);
+
+      for (let survey of uniqueServeys) {
+        await Survey.updateOne(
+            {
+              _id: survey.surveyId,
+              recipients: {
+                $elemMatch: {email: survey.email, responded: false}
+              }
+            },
+            {
+              $inc: {[survey.choice]: 1},
+              $set: {'recipients.$.responded': true},
+              lastResponded: new Date()
+            }
+        ).exec();
+      }
```

Comment:
If we weren't using async/await for this, I would just chain a `.forEach` onto my javascript array chain, which would look nice:

```
data.map(this.buildSurveyDataFromBody)
  .filter(this.filterOutNullAndUndefined)
  .filter(this.filterByUniqueEmailAndSurveyId)
  .forEach(this.writeSurveysToDatabase);
```

But `forEach` doesn't work with async/await, so I had to use a for/of loop instead. This mean the loop waits until the database write is complete before continuing, and then this function won't return until all writes are complete.